docs/setup.md

```diff
@@ -0,0 +1,51 @@
+# Setup Script Configuration
+
+This document provides a guide on how to use the `bin/setup` script for setting up your RapidRails application. The script automates several setup tasks, including installing system and Ruby dependencies, setting up Rails credentials and environment variables, preparing the database and more.
+
+## Services Required
+
+Before running the script, ensure you have the following services installed:
+
+- Ruby
+- Homebrew (for macOS users)
+- Bundler
+
+## Script Features
+
+The script performs the following tasks:
+
+1. System Dependencies Installation: If Homebrew is installed, the script will install system dependencies using the brew bundle command.
+
+2. Ruby Dependencies Installation: The script installs Ruby dependencies using Bundler.
+
+3. Rails Credentials & ENV Variables Setup: The script prepares the .env file and Rails credentials file.
+
+4. Database Setup: The script prepares the database.yml file and sets up the database using Rails commands.
+
+5. Other Tasks: The script also removes old logs and temp files, and restarts the application server.
+
+## Manual Setup
+
+If you prefer to perform the setup manually, follow these steps:
+
+1. System Dependencies: Install system dependencies manually using Homebrew or your preferred package manager.
+
+2. Ruby Dependencies: Run bundle install to install Ruby dependencies.
+
+3. Rails Credentials & ENV Variables: Create a .env file from the .env.template file. For Rails credentials, create config/credentials.yml.enc and config/master.key files manually.
```

Comment:
say the command for this. 

so if this is a new codebase then they would delete the current encrypted file and run the command the script runs to create a new one. 

if it's not then tell them that they need to add the master.key file

Reply:
actually for this one, we have a separate file that goes through this in detail. maybe you can just link to that from here. https://github.com/danielpaul/RapidRails/pull/56/files